### PR TITLE
Added POST featurs

### PR DESCRIPTION
**Pull request by MarkR92**
Added POST feature. Very basic. Currently can handle gets and posts .

### CODE REVIEW

File: ServerSocketTest/src/main/Main.java [hunk 1]
```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
...
+							
+							body+=""+(char)buff.read();//check the next char
+							
+							
+						}
+						
+						Pattern pattern2 = Pattern.compile("person\\d=([^&]+)");
+
+				        // Create a matcher with the input string
+				        Matcher matcher2 = pattern2.matcher(body);
```
Comment (Robmurt4) [inline ServerSocketTest/src/main/Main.java:172]: What does the  matcher() class do in this situation?

File: ServerSocketTest/src/main/Main.java [hunk 2]
```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
...
+						
+						clientOutput.flush();//empty the built up buffer
+					}
+					else if(resource.equals("/"))//the root doesnt do anything rn. Just a "welcome page"
+					{
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						
+						clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						clientOutput.write(("Hello").getBytes());//encode to bytes
```
Comment (Robmurt4) [inline ServerSocketTest/src/main/Main.java:114]: What if we want to pull multiple requests from the server. Would we have to do .write(("HTTP/1.1 200 OK\r\n").getBytes() and .write(("\r\n").getBytes()  for every message we want to pull or just once at the start.

File: ServerSocketTest/src/main/Main.java (hunk 1 above)
Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:172]: It "counts" the number of matches found in a string (the body)  that correspond to the regex pattern. I should have reassigned pattern and matcher as opposed to create new variables matcher2 partten2 etc

Basically I want to find number of people in the body of the POST and add them to and Array

File: ServerSocketTest/src/main/Main.java [hunk 3]
```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
...
-					clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
-					clientOutput.write(("\r\n").getBytes());//blank line
-					clientOutput.write(("Hello").getBytes());//encode to bytes
+									
+					if(resource.equals("/people"))
+					{
+						int contentLength=0;
+						//parse out content length using the much beloved regex
+						   // Define the pattern to match "Content-Length: value"
+				        Pattern pattern = Pattern.compile("Content-Length: (\\d+)");
```
Comment (Robmurt4) [inline ServerSocketTest/src/main/Main.java:140]: I'm confused about the whole pattern stuff.

File: ServerSocketTest/src/main/Main.java (hunk 2 above)
Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:114]: Each request has a response. So I would say yes to this.

Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:114]: Just to clarify we will wrap this in a method and just call the method every time we have a 200 OK response

File: ServerSocketTest/src/main/Main.java [hunk 4]
```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
 				
 				
 				//get the first line of the request
-				String firstLine=request.toString().split("\n")[0];
+				String firstLine=header.toString().split("\n")[0];
 				//System.out.println(firstLine);
 				//get the second element from first line
 				String resource=firstLine.split(" ")[1];
```
Comment (Robmurt4) [inline ServerSocketTest/src/main/Main.java:80]: I'm confused where the value of resource comes from I see it comes from the firstLine string but how does it get the value for "people".

File: ServerSocketTest/src/main/Main.java (hunk 3 above)
Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:140]: Content-Length is apart of the header of a request and an example of a header from a post request is:
POST /people HTTP/1.1
Host: localhost:8097
User-Agent: curl/8.4.0
Accept: */*
Content-Length: 65
Content-Type: application/x-www-form-urlencoded


We need to parse out the Content-Length to find the value 65 (headers can have many lines above is just an example). There are many ways to parse this information. One of these ways is to search the header for a pattern. I define this pattern using the built in pattern class and regular expression .

So line 140 means search the header until you reach the line containing "Content-Length" once we find this line extract the data contained within the brackets () which will be a number of various length. I need this number so I know how much info is in the body of the request.

File: ServerSocketTest/src/main/Main.java (hunk 4 above)
Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:80]: When we make a GET request from curl ( enter the following into command line: curl localhost:8097/people ) we get the following:
GET /people HTTP/1.1 <----- there is a hidden \n here
Host: localhost:8097
User-Agent: curl/8.4.0
Accept: */*

line 77 pulls out the first line of the header which is:
GET /people HTTP/1.1
line 80 splits everywhere we see a space which results in:
GET
/people
HTTP/1.1

and specifying [1] gets the second element from the split which I call the resource

File: ServerSocketTest/src/main/Main.java [hunk 5]
```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
...
+						
+						Pattern pattern2 = Pattern.compile("person\\d=([^&]+)");
+
+				        // Create a matcher with the input string
+				        Matcher matcher2 = pattern2.matcher(body);
+
+				        // Find the names this only works for the following format person1=mark&person2=rob&person3=khate&person4=dan&person5=dorian
+				        while (matcher2.find()) {
+				            String name = matcher2.group(1);
+				            people.add(name);//encode to bytes
```
Comment (Danielius-Simonavicius) [inline ServerSocketTest/src/main/Main.java:177]: So this logic is something a particular website hosted under your web server would do. Your web server simply needs to pass the request to website?! If it's just a test, then ignore my comment of course.

Comment (rmarmu) [reply ServerSocketTest/src/main/Main.java:177]: Yes. except the website passes the request to web server and the webserver responds. This is a prototype/reference of how we program a POST. I completely made up the data to be passed into the arrayList.

**Review by Robmurt4 — APPROVED** [review]

**Pull request merged**